internal/proxyattributes/proxyattributes_test.go

```diff
@@ -0,0 +1,119 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package proxyattributes
+
+import (
+	"net/url"
+	"testing"
+
+	"google.golang.org/grpc/attributes"
+	"google.golang.org/grpc/internal/grpctest"
+	"google.golang.org/grpc/resolver"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+// Tests that ConnectAddr returns the correct connect address in the attribute.
+func (s) TestConnectAddr(t *testing.T) {
+	tests := []struct {
+		name string
+		addr resolver.Address
+		want string
+	}{
+		{
+			name: "connect address in attribute",
```

Comment:
Please use underscores in subtest names. I'm sorry if I asked you to do the opposite in the past. I also learned this recently from Arjan.

See: https://g3doc.corp.google.com/company/teams/go-community/gotip/episodes/117.md?cl=head